core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -513,6 +523,16 @@ public SliceBuilder slice() {
         return sliceBuilder;
     }
 
+
+    public CollapseBuilder collapse() {
```

Comment:
Ah sorry, I see, but can take `collapse` into account in the hashcode and equals method of SearchSourceBuilder?